primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectoneradio.js

```diff
@@ -174,6 +174,21 @@ PrimeFaces.widget.SelectOneRadio = PrimeFaces.widget.BaseWidget.extend({
             e.preventDefault();
         });
 
+        // for custom radio buttons, transfer the focus from original to the matching custom button
+        $(this.originalInputs).on('focus.selectOneRadio', function() {
+            // 1 = JSF separator, 2 = index
+            var matches = this.id.match(/^.*([^\d])(\d)+$/);
```

Comment:
please use seperatorChar instead of regex like discussed

Reply:
I've started over to address the Safari issue first. I won't be using the regex.